DiegoFigueroaSevillano/Borbotones-FinalProject
# feat: thread implementation

### PR Description

This PR enabling concurrent client handling and simulating high-demand scenarios using threads. Updates:

Server-Side Multi-Client Handling:

BorbotonesSystemController now leverages threads to manage each client connection independently.
Each incoming connection spawns a new thread to handle client requests, ensuring the server can process multiple clients simultaneously.

Client Simulation of High Demand:

Added two client classes (Client1 and Client2) that continuously send requests to the server at 2-second and 3-second intervals, respectively.
Each client reads from its specific text files and performs text comparison using the TextComparatorService.

Enhanced Logging:

The server logs incoming connections, processed requests, and results sent back to the clients.
Clients log their request numbers to track interactions with the server.

### How to Test
Start the Server:

Execute `Service.Main` class it to start the server:
Confirm that the server is running by checking the console output:
`Server is running on port 8080.`
Start Client 1:

Execute `Client1`:
Observe that the client begins sending requests every 2 seconds.
Logs will show the paths sent and the results received.
Start Client 2:

 Execute `Client2`:
Observe that this client sends requests every 3 seconds.
Logs will show its interactions with the server independently of Client 1.
Verify Concurrent Processing:

Check the server logs to ensure it is processing requests from both clients concurrently.
The server should display logs of Client 1 and Client 2 activity.
